geonode/people/views.py

```diff
@@ -189,6 +192,50 @@ def get_queryset(self):
         queryset = self.get_serializer_class().setup_eager_loading(queryset)
         return queryset.order_by("username")
 
+    def perform_destroy(self, instance):
+        # not implemented added to make tests pass
+        if any(
+            (
+                not self.request.user.is_superuser,
+                not self.request.user.is_staff,
+                self.request.user.pk == int(self.kwargs["pk"]),
```

Comment:
THis changes are needed, but i guess is better to move them into a `PermissionsClasses` . by adding this (and their needed checks) the changes done in the `IsSelf` are no longer needed and we can remove it from the permissions list of the API